File: Year 2/Semester 4/Programming and Projection Enironments/Week 8/SeventhWeekCSharpServerClient/SeventhWeekCSharpServerClient/StartRPCServer.cs
using System;
using System.Collections.Generic;
using System.Configuration;
using SeventhWeekCSharpServerClient.repository;
using SeventhWeekCSharpServerClient.repository.interfaces;
using SeventhWeekCSharpServerClient.services;
using SeventhWeekCSharpServerClient.networking.utils;

namespace SeventhWeekCSharpServerClient.server
{
    internal static class StartRPCServer
    {
        [STAThread]
        private static string GetConnectionStringByName(string name)
        {
            string returnValue = null;
            var settings = ConfigurationManager.ConnectionStrings[name];
            if (settings != null)
                returnValue = settings.ConnectionString;
            return returnValue;
        }

        private static void Main(string[] args)
        {

            IDictionary<string, string> props = new SortedList<string, string>();
            props.Add("ConnectionString", GetConnectionStringByName("shows"));
            IArtistRepository artistRepository = new ArtistDBRepository(props);
            IEmployeeRepository employeeRepository = new EmployeeDBRepository(props);
            IShowRepository showRepository = new ShowDBRepository(props);
            ITicketRepository ticketRepository = new TicketDBRepository(props);
            IServices service = new ServicesImpl(artistRepository, employeeRepository, showRepository, ticketRepository);
            var server = new RpcConcurrentServer("127.0.0.1", 55556, service);
            try
            {
                server.Start();
                Console.WriteLine(@"Server started ...");
                Console.ReadLine();
            }
            catch (ServerException e)
            {
                Console.Error.WriteLine("Error starting the server" + e.Message);
            }
            finally
            {
                try
                {
                    server.Stop();
                }
                catch (ServerException e)
                {
                    Console.Error.WriteLine("Error stopping server " + e.Message);
                }
            }
            
        }
    }

}

File: Year 2/Semester 4/Programming and Projection Enironments/Week 10/ProtobuffProjectJavaORM/ServerRPC/src/main/java/StartRpcServer.java
import gRPCProject.domain.Artist;
import gRPCProject.domain.Employee;
import gRPCProject.domain.Show;
import gRPCProject.domain.Ticket;
import gRPCProject.network.utils.AbstractServer;
import gRPCProject.network.utils.RpcConcurrentServer;
import gRPCProject.network.utils.ServerException;
import gRPCProject.persistance.repository.IArtistRepository;
import gRPCProject.persistance.repository.IEmployeeRepository;
import gRPCProject.persistance.repository.IShowRepository;
import gRPCProject.persistance.repository.ITicketRepository;
import gRPCProject.persistance.repository.jdbc.ArtistDBIRepository;
import gRPCProject.persistance.repository.jdbc.EmployeeDBIRepository;
import gRPCProject.persistance.repository.jdbc.ShowDBIRepository;
import gRPCProject.persistance.repository.jdbc.TicketDBIRepository;
import gRPCProject.server.ServicesImpl;
import gRPCProject.service.IService;

import java.io.IOException;
import java.util.Properties;

public class StartRpcServer {
    private static int defaultPort=55555;

    public static void main(String[] args) {
        // UserRepository userRepo=new UserRepositoryMock();
        Properties serverProps=new Properties();
        try {
            serverProps.load(StartRpcServer.class.getResourceAsStream("/server.properties"));
            System.out.println("Server properties set. ");
            serverProps.list(System.out);
        } catch (IOException e) {
            System.err.println("Cannot find server.properties "+e);
            return;
        }
        IArtistRepository<Integer, Artist> artistRepository = new ArtistDBIRepository(serverProps);
        IEmployeeRepository<Integer, Employee> employeeRepository = new EmployeeDBIRepository(serverProps);
        IShowRepository<Integer, Show> showRepository = new ShowDBIRepository(serverProps);
        ITicketRepository<Integer, Ticket> ticketRepository = new TicketDBIRepository(serverProps);
        IService serverImpl=new ServicesImpl(artistRepository, employeeRepository, showRepository, ticketRepository);

        int serverPort=defaultPort;
        try {
            serverPort = Integer.parseInt(serverProps.getProperty("server.port"));
        }catch (NumberFormatException nef){
            System.err.println("Wrong  Port Number"+nef.getMessage());
            System.err.println("Using default port "+defaultPort);
        }
        System.out.println("Starting server on port: "+serverPort);
        AbstractServer server = new RpcConcurrentServer(serverPort, serverImpl);
        try {
            server.start();
        } catch (ServerException e) {
            System.err.println("Error starting the server" + e.getMessage());
        }finally {
            try {
                server.stop();
            }catch(ServerException e){
                System.err.println("Error stopping server "+e.getMessage());
            }
        }
    }
}
